static_frame/core/frame.py

```diff
@@ -832,6 +832,15 @@ def __init__(self,
             # will have shape of 0,0
             self._blocks = TypeBlocks.from_none()
 
+        elif data is None and (
+                hasattr(index, '__len__') or hasattr(columns, '__len__')):
+            index_length = len(index) if hasattr(index, '__len__') else 0
+            column_length = len(columns) if hasattr(columns, '__len__') else 0
+            # Will have shape (len of index, length of columns).
+            a = np.full((index_length, column_length), None)
```

Comment:
We can probably remove the `data is None` check on 835, and just do this here:

```suggestion
            a = np.full((index_length, column_length), data)
```

That way the `data` value can be used to set the `dtype` of the empty array!